packages/airnode-deployer/src/handlers/aws/index.ts

```diff
@@ -29,9 +30,30 @@ export async function initializeProvider(event: any) {
   return { statusCode: 200, body };
 }
 
-export async function callApi(event: any) {
-  const { aggregatedApiCall, logOptions, encodeResponse } = event;
-  const [logs, apiCallResponse] = await handlers.callApi(parsedConfig, aggregatedApiCall, encodeResponse);
+function parseCallApiPayload(event: any): handlers.CallApiPayload {
+  const requiredKeys = ['config', 'aggregatedApiCall', 'apiCallOptions'];
+
+  // NOTE: We could do more extensive checks, but key renames are the most common cause of failure
+  requiredKeys.forEach((key) => {
+    if (Object.prototype.hasOwnProperty.call(event, key)) {
+      throw new Error(`Parsing payload for callApi failed. Event is missing required property ${key}`);
+    }
+  });
+
+  return pick(event, requiredKeys) as handlers.CallApiPayload;
+}
+
+export async function callApi(event: unknown) {
+  // TODO: Verify that logOptions are present
+  const { logOptions } = event as any;
+
+  const [error, callApiPayload] = promiseUtils.goSync(() => parseCallApiPayload(event));
```

Comment:
@andreogle @amarthadan LMK WDYT about this...

All of these AWS handlers are "black magic" boxes to me. I find it dangerous to expect that the `event` will contain the expected properties (especially after we do a bunch of refactors). I think we should check for them explicitly...

Also, I still don't understand completely what is contained in the `event` object (especially how `config` ges there in this example).

Reply:
Yeah I agree. I like this.

Reply:
I agree that checking the content of the `event` object is a good idea but wouldn't it be better to do it on the `airnode-node` side instead of `airnode-deployer`? In that case code would be only at one place even for different cloud providers.

Reply:
> Also, I still don't understand completely what is contained in the event object (especially how config ges there in this example).

I'm not sure what you mean. The config is parsed from `config.json` (line 5).

Reply:
This is my bad - I wrongly refactored the code and then it made no sense to me... I have removed the refactor parts and will improve that in a separate PR